app/controllers/user_activations_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class UserActivationsController < ApplicationController
+  include UsersHelper
+
+  def activate_form
+    assign_user(params[:id])
+    check_activation_permissions
+    redirect_to users_url if @user.active?
+  end
+
+  def deactivate_form
+    assign_user(params[:id])
+    check_activation_permissions
+    redirect_to users_url unless @user.active?
+  end
+
+  def activate
+    assign_user(params[:id])
+    check_activation_permissions
+    @user.activate! unless @user.active?
+    redirect_to users_url
+  end
+
+  def deactivate
+    assign_user(params[:id])
+    check_activation_permissions
```

Comment:
I do believe that stuff like `check_activation_permissions` should always be on top of the class as `before_action` rather than called in the methods

Reply:
Yep, can't remember why I have done this. Fixed!